optuna/integration/botorch.py

```diff
@@ -238,6 +238,95 @@ def qehvi_candidates_func(
     return candidates
 
 
+@experimental_func("2.4.0")
+def qnehvi_candidates_func(
+    train_x: "torch.Tensor",
+    train_obj: "torch.Tensor",
+    train_con: Optional["torch.Tensor"],
+    bounds: "torch.Tensor",
+) -> "torch.Tensor":
+    """Quasi MC-based batch Expected Noisy Hypervolume Improvement (qnehvi).
+
+    TODO: [some documentation]
+
+    .. seealso::
+        :func:`~optuna.integration.botorch.qei_candidates_func` for argument and return value
+        descriptions.
+    """
+
+    n_objectives = train_obj.size(-1)
+
+    if train_con is not None:
+        train_y = torch.cat([train_obj, train_con], dim=-1)
+
+        # is_feas = (train_con <= 0).all(dim=-1)
+        # train_obj_feas = train_obj[is_feas]
+
```

Comment:
```suggestion
```
Remove unnecessary comments.

Reply:
The code has been modified as the comment.
https://github.com/eukaryo/optuna/commit/d88d655ae0dc1f2bf0d974cef097ae6d36dab2d4